packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5542,8 +5555,13 @@ class RawImage extends LeafRenderObjectWidget {
 
   @override
   void updateRenderObject(BuildContext context, RenderImage renderObject) {
+    assert(
+      image?.debugGetOpenHandleStackTraces()?.isNotEmpty ?? true,
+      'Creator of a RawImage disposed of the image when the RawImage still '
+      'needed it.'
+    );
     renderObject
-      ..image = image
+      ..image = image?.clone()
```

Comment:
why does this clone if it doesn't dispose? won't this create extra image clones

Reply:
The render object disposes of it. This can't dispose because it's stateless - it needs to be kept alive by whatever created RawImage.

Reply:
This is pretty confusing - so the State class has its copy, which it will clone/dispose as needed. Then the RawImage is just a pass through, and finally the RO accepts the clone and does a dispose.

Maybe a safer arrangement would be for the RO to clone in the setter, so that the behavior is internal to it?

Reply:
I'm trying to keep the contract as you should never have to clone something you received. If we have some places that clone immediately I think it's more confusing.

However, I agree that `RawImage` is a sore spot, and I think it'd be preferable to just make it private since it's easy to misuse.

Reply:
I have also been confused by the fact that cloning and disposing are done by separate entities.

Reply:
Yeah, if it needs to act like an implementation detail of the Image class it should be private. But we do have other `RawFoo` classes exposed in the framework so that people could build on top of that, though I think there is also a rendering helper method too?

I dunno.

Reply:
Refactored RawImage to not be a special case. There's a private impl that's a special case now though.

Reply:
I had to revert that refactor because it doesn't work to have the image hold a stateful object like that.

Reply:
why not have the widget pass through, and its caller (the call to the RawImage constructor) have to call `clone()`?

Reply:
The widget might create multiple RenderObjects (via multiple elements). It has no defined lifetime, and so it's not clear when the widget's clone would ever get disposed.